tests/suites/test_suite_opaque_test_driver.function

```diff
@@ -0,0 +1,643 @@
+    psa_key_attributes_t attr = PSA_KEY_ATTRIBUTES_INIT;
+
+    TEST_ASSERT( PSA_SUCCESS == psa_crypto_init() );
+
+    psa_set_key_type( &attr, type_arg );
+    psa_set_key_bits( &attr, bits_arg );
+
+    TEST_ASSERT( PSA_SUCCESS == opaque_test_driver_generate_key( &attr,
+                                                                 key,
+                                                                 OPQ_BUFSIZE,
+                                                                 &key_len ) );
+exit:
+    psa_reset_key_attributes( &attr );
+    PSA_DONE( );
+    #undef OPQ_BUFSIZE
+}
+/* END_CASE */
+
+/* BEGIN_CASE */
+void opaque_test_driver_import_key( char *key_material )
+{
+    #define OPQ_BUFSIZE ( 32U + OPAQUE_TEST_DRIVER_KEYHEADER_SIZE )
+    size_t out_key_len = 0;
+    psa_key_attributes_t attr = PSA_KEY_ATTRIBUTES_INIT;
+    uint8_t out_key1[OPQ_BUFSIZE + 1];
+    uint8_t out_key2[OPQ_BUFSIZE + 1];
+
+    TEST_ASSERT( PSA_SUCCESS == psa_crypto_init() );
+
+    psa_set_key_type( &attr, PSA_KEY_TYPE_AES );
+    psa_set_key_bits( &attr, PSA_BYTES_TO_BITS( strlen( key_material) ) );
+
+    TEST_ASSERT( strlen( key_material ) <= 32 );
+
+    /* Import key. */
+    TEST_ASSERT( PSA_SUCCESS == opaque_test_driver_import_key(
+                                    &attr,
+                                    (uint8_t *) key_material,
+                                    strlen( key_material ),
+                                    out_key1,
+                                    OPQ_BUFSIZE + 1,
+                                    &out_key_len ) );
+
+    TEST_ASSERT( strlen( key_material ) + OPAQUE_TEST_DRIVER_KEYHEADER_SIZE ==
+                 out_key_len );
+    out_key1[out_key_len] = '\0';
+
+    /* Import the "imported" key (less key header). */
+    TEST_ASSERT( PSA_SUCCESS == opaque_test_driver_import_key(
+                                    &attr,
+                                    out_key1 +
+                                        OPAQUE_TEST_DRIVER_KEYHEADER_SIZE,
+                                    strlen( (char *) out_key1 +
+                                        OPAQUE_TEST_DRIVER_KEYHEADER_SIZE ),
+                                    out_key2,
+                                    OPQ_BUFSIZE + 1,
+                                    &out_key_len ) );
+
+    TEST_ASSERT( strlen( (char *) out_key1 ) == out_key_len );
+    out_key2[out_key_len] = '\0';
+
+    /* The keys should be equal when ignoring key header. */
+    TEST_ASSERT( 0 == mbedtls_test_hexcmp( (uint8_t *) key_material,
+                          out_key2 + OPAQUE_TEST_DRIVER_KEYHEADER_SIZE,
+                          strlen( key_material ),
+                          strlen( (char *) out_key2 +
+                              OPAQUE_TEST_DRIVER_KEYHEADER_SIZE ) ) );
+exit:
+    psa_reset_key_attributes( &attr );
+    PSA_DONE( );
+    #undef OPQ_BUFSIZE
+}
+/* END_CASE */
+
+/* BEGIN_CASE depends_on:MBEDTLS_CHECK_PARAMS:!MBEDTLS_PARAM_FAILED_ALT */
```

Comment:
`depends_on:MBEDTLS_CHECK_PARAMS` has no effect on PSA APIs, only on traditional Mbed TLS APIs. (It was added because traditional APIs didn't perform certain checks, but the PSA code either performs those checks or has an interface that makes them moot.) This test function is not useful.